lib/p2p/Pool.ts

```diff
@@ -987,6 +1016,35 @@ class Pool extends EventEmitter {
       }
     });
   }
+
+  /**
+   * Resolves alias to node's public key
+   */
+  private resolveAlias = (alias: string) => {
+    const plist = this.listPeers();
```

Comment:
The aliases are only used in the CLI and for logging, and so I don't think they are needed for all nodes. I am implementing all your other suggestions but not this one due to the additional complexity of assigning an alias to all nodes, not just to peers.

Reply:
I.e. aliases are only used for CLI actions effecting connected nodes (peers)

Reply:
I agree that the aliases system would be improved by allowing specification of nodes that are not connected. This will actually be a small change now that I think it through:

Ban seems more like a debugging/testing feature, since ideally there will be automatic banning of nodes once a low-enough reputation is earned. I cannot imagine any users sitting at the CLI and telling each other to ban a specific alias and thereby benefitting from the ease-of-use of aliases. Any malicious node operator can simply start a new node or generate a new public key if they get banned for social reasons, i.e. not automatically earned reputation points. Unban via alias will work as-is since it looks for the node in the list of banned nodes. This is a valuable feature because users can enjoy easily unbanning a banned node that they want to connect to. This leaves getnodeinfo, which should allow the user to get information about any node in the system. The function that collects this info looks at the list of all nodes, and therefore I can make a small change to just this function so that the aliases will work as expected.